src/item_factory.cpp

```diff
@@ -134,11 +134,13 @@ void Item_factory::finalize() {
             }
 
             // if the gun doesn't have a DEFAULT mode then add one now
-            obj.gun->modes.emplace( "DEFAULT", std::tuple<std::string, int, std::set<std::string>>( defmode, 1, {} ) );
+            obj.gun->modes.emplace( "DEFAULT", std::tuple<std::string, int, std::set<std::string>>( defmode, 1,
+                                    std::set<std::string>() ) );
```

Comment:
Possible solution for mitigating ugliness would be abandoning `tuple`s (and `pair`s). They're quite convenient for templates, but if templates aren't used here, why use `tuple`s? Explicit `struct` is much cleaner than a bunch of "anonymous" fields.